satellite/gracefulexit/endpoint.go

```diff
@@ -327,31 +337,27 @@ func (endpoint *Endpoint) doProcess(stream processStream) (err error) {
 
 		switch m := request.GetMessage().(type) {
 		case *pb.StorageNodeMessage_Succeeded:
-			err = endpoint.handleSucceeded(ctx, pending, nodeID, m)
+			err = endpoint.handleSucceeded(ctx, stream, pending, nodeID, m)
 			if err != nil {
 				if ErrInvalidArgument.Has(err) {
 					// immediately fail and complete graceful exit for nodes that fail satellite validation
-					// TODO(moby) use getFinishedMessage helper after 3368 is merged
 					exitStatusRequest := &overlay.ExitStatusRequest{
 						NodeID:         nodeID,
 						ExitFinishedAt: time.Now().UTC(),
 						ExitSuccess:    false,
 					}
-					// TODO needs signature
-					transferMsg := &pb.SatelliteMessage{
-						Message: &pb.SatelliteMessage_ExitFailed{
-							ExitFailed: &pb.ExitFailed{
-								Reason: pb.ExitFailed_VERIFICATION_FAILED,
-							},
-						},
+
+					finishedMsg, err := endpoint.getFinishedMessage(ctx, endpoint.signer, nodeID, *exitStatus.ExitFinishedAt, exitStatus.ExitSuccess, pb.ExitFailed_VERIFICATION_FAILED)
```

Comment:
I think the panic is coming from here because the arguments need to use `exitStatusRequest.ExitFinishedAt` and `exitStatusRequest.ExitSuccess` instead of `exitStatus`